source/dub/internal/utils.d

```diff
@@ -130,6 +130,39 @@ Json jsonFromFile(NativePath file, bool silent_fail = false) {
 	return parseJsonString(text, file.toNativeString());
 }
 
+/**
+	Read package info file content from archive.
+	File needs to be in root folder or in first
+	sub folder.
+
+	Params:
+		zip = path to archive file
+		fileName = Package file name
+	Returns:
+		package file content.
+*/
+string packageInfoFileFromZip(NativePath zip, out string fileName) {
+	import std.zip : ZipArchive, ArchiveMember;
+	import dub.package_ : packageInfoFiles;
+
+	auto f = openFile(zip, FileMode.read);
+	ubyte[] b = new ubyte[cast(size_t)f.size];
+	f.rawRead(b);
+	f.close();
+	auto archive = new ZipArchive(b);
+	alias PSegment = typeof (NativePath.init.head);
+	outer: foreach (ArchiveMember am; archive.directory) {
```

Comment:
Why is the outer label doing there>

Reply:
Good catch:)
It is superfluous and was needed in a previous implementation, I will remove it.

Reply:
Done